src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,78 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final int dateIndex = 0;
+    private static final int nameIndex = 1;
+    private static final int hoursWorkedIndex = 2;
+    private static final int hourlyRateIndex = 3;
+    private static final int expectedParts = 4;
+    private static final DateTimeFormatter dateFormatter
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    private boolean isValidEntry(String[] parts, String name,
+                                 LocalDate fromDate, LocalDate toDate) {
+        if (parts.length == expectedParts) {
+            try {
+                LocalDate entryDate = LocalDate.parse(parts[dateIndex], dateFormatter);
+                return !entryDate.isBefore(fromDate)
+                        && !entryDate.isAfter(toDate)
+                        && parts[nameIndex].equals(name);
+            } catch (DateTimeParseException e) {
+                return false;
+            }
+        }
+        return false;
```

Comment:
don't keep a lot of code in an if-block
```suggestion
        if (parts.length != expectedParts) {
           return false;
       }
            try {
                LocalDate entryDate = LocalDate.parse(parts[dateIndex], dateFormatter);
                return !entryDate.isBefore(fromDate)
                        && !entryDate.isAfter(toDate)
                        && parts[nameIndex].equals(name);
            } catch (DateTimeParseException e) {
                return false;
            }
```